tests/rptest/services/admin.py

```diff
@@ -533,6 +537,26 @@ def _request(self,
             self.redpanda.logger.debug(f"Dispatching {verb} {url}")
             try:
                 r = self._session.request(verb, url, **kwargs)
+                # Requests library does NOT respect Retry-After with a 307
+                # error code (see https://github.com/psf/requests/pull/4562).
+                # There is logic that respects Retry-After within urllib3,
+                # but since the Requests library handles 30# error codes
+                # internally, a Retry-After attached to a redirect response
+                # will not be respected. We will have to handle this ourselves.
+                while r.status_code == 307 and (
+                        num_redirects := 1) < self._session.max_redirects:
+                    redirect_url = r.headers.get('Location')
+                    retry_after = r.headers.get('Retry-After')
+                    if retry_after is not None:
+                        self.redpanda.logger.info(
```

Comment:
I'm curious if this works with rpk as expected.

Reply:
I think we may have to add our own `Retry-After` parsing to `httpapi/httpapi.go` in rpk to ensure the backoff is respected.

Reply:
thanks for checking..we should probably create an issue for devex team?